ql/src/java/org/apache/hadoop/hive/ql/exec/tez/TezSessionState.java

```diff
@@ -317,7 +317,8 @@ protected void openInternal(final HiveConf conf, Collection<String> additionalFi
 
     setupSessionAcls(tezConfig, conf);
 
-    final TezClient session = TezClient.newBuilder("HIVE-" + sessionId, tezConfig)
+    String tezJobName = HiveConf.getVar(conf, ConfVars.HIVETEZJOBNAME, sessionId);
+    final TezClient session = TezClient.newBuilder(String.format("HIVE-%s", tezJobName), tezConfig)
```

Comment:
Here is what I am suggesting:

```
    String tezJobNameFormat = HiveConf.getVar(conf, ConfVars.HIVETEZJOBNAME);
    final TezClient session = TezClient.newBuilder(String.format(tezJobNameFormat, sessionId), tezConfig)

```

Reply:
Thanks for the hint, I have updated the code according to your suggestion